protocol/test/forge/v2/interestRate/LinearWithKinkInterestRateModel.t.sol

```diff
@@ -0,0 +1,165 @@
+pragma solidity ^0.8.17;
+// SPDX-License-Identifier: AGPL-3.0-or-later
+import {ud} from '@prb/math/src/UD60x18.sol';
+
+import {TempleTest} from '../../TempleTest.sol';
+import {LinearWithKinkInterestRateModel} from 'contracts/v2/interestRate/LinearWithKinkInterestRateModel.sol';
+import {CommonEventsAndErrors} from 'contracts/common/CommonEventsAndErrors.sol';
+import {IInterestRateModel} from 'contracts/interfaces/v2/interestRate/IInterestRateModel.sol';
+import {FakeERC20} from 'contracts/fakes/FakeERC20.sol';
+import { SafeCast } from "contracts/common/SafeCast.sol";
+
+contract Mock is LinearWithKinkInterestRateModel {
```

Comment:
Hmm why do you need a mock contract? You can just create a LinearWithKink as the thing to test below

Reply:
If I want to test the `setRateParams` function and their access, I either need to use a mock contract, or add the function to the interface `IInterestRateModel`, which currently only have the `calculateInterestRate`, so that would be the only function I could test afaik, unless there is another way I'm not aware? 

I'll give it a go for now by adding `setRateParams` to `IInterestRateModel`